alica_ros_turtlesim/README.md

```diff
@@ -0,0 +1,196 @@
+ALICA  manage state transitions based on user code. The alica-plan-designer generates code strucutre and developer implement state transition logics. 
+Developer can create plan using alica-plan-designer.
+##### Behaviour
+Developer can write robot behaviour at each state in C++ code. alica-plan-designer generate code structure and developer implement behaviour logics. In this tutorial, there are `Go2RandomPosition` and `GoTo`.
+Turtles are teleported to random position with `Go2RandomPosition` and 
+Role: Role is task preference of the agent. Role describes physical difference among agents , e.g. robot arm and AGV. In this tutorial all agent have same `Turtle` Role.
+##### Task
+Task is assign  based on `Role` and `Utility function` by ALICA. Tassk is entry point of state machine. ALICA realize multi agent collaboration by assigning Task to agent. In this tutorial, there are `Leader task` and `Follower task`. One turtle is assigned `Leader` and move to center, other turtles are assigned `Follower` and align circle.
+##### Constraints
+Developer can set constraints to the plans. ALICA can solve constraints and return answer. In this tutorial, turtles align circle by constraints.
+Worldmodel: World model represents agent model in the world. World model is a interface between ALICA and other software, e.g. ROS and lower API.
+
+![coreconcept](https://github.com/rapyuta-robotics/alica-supplementary/blob/add_turtlesim/alica_ros_turtlesim/doc/coreconcept.png)
+
+### 4. Set up workspace
+Creating workspace by following steps in Ubuntu16.04 terminal.
+```
+mkdir -p catkin_ws/src
+cd catkin_ws/src
+git clone https://github.com/rapyuta-robotics/alica.git
+git clone https://github.com/rapyuta-robotics/alica-essentials.git
+git clone https://github.com/rapyuta-robotics/alica-supplementary.git
+
+# remove existing turtlesim files. You will reproduce these files.
+# you can jump to 8. "Build and Run" for testing application before deleting
+cd src/alica-supplementary
+rm -r alica_ros_turtlesim/Expr 
+rm -r alica_ros_turtlesim/alica
+
+catkin_create_pkg alica_ros_turtlesim roscpp turtlesim geometry_msgs alica_engine agent_id alica_ros_proxy constraintsolver
+mkdir -p alica_ros_turtlesim/alica/etc/plans/behaviours
+mkdir -p alica_ros_turtlesim/alica/etc/roles
+mkdir -p alica_ros_turtlesim/alica/etc/Misc
+mkdir -p alica_ros_turtlesim/Expr
+```
+
+### 5. Set up alica plan designer
+Plan designer is a user interface to design application with ALICA.
+##### Start alica plan designer by following steps
+
+```
+cd catkin_ws/src/alica-plan-designer
+./bin/start.sh
+```
+![plan-designer_project_config](https://github.com/rapyuta-robotics/alica-supplementary/blob/add_turtlesim/alica_ros_turtlesim/doc/plan-designer_project_config.png)
+##### Plan designer setting
+Create project: input root path to your alica ws and change expr as well.
+#todo add image
+Window -> Preference -> Plan Designer -> Condition plugin
```

Comment:
Path to plugin folder shall be : 
<path_to_plan_designer>/alica-plan-designer/src/conditionPlugins/plugins

Reply:
otherwise, code generation issues.